### ISSUE TITLE
Synchronize: a problem with a non-standard ssh port

### ISSUE THREAD

**Issue by dvzubarev** [ansible/ansible-modules-core#3895]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

synchronize
##### ANSIBLE VERSION

```
<!--- Paste verbatim output from “ansible --version” between quotes -->
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

[defaults]
vault_password_file=.vault_pass.txt
##### OS / ENVIRONMENT

Ubuntu 16.04
##### SUMMARY

I have the following option in my /etc/ssh/ssh_config : 

```
 Port 2222
```

I want to sync some directory to a machine that has this address: 10.0.3.188:22.

I get an error 

```
 ssh: connect to host 10.0.3.188 port 2222: Connection refused
```

This [code](https://github.com/ansible/ansible-modules-core/blob/c52f475c64372042daab4ebc0660a2782b71d10d/files/synchronize.py#L414-L417) is responsible for this behavior. Why don't set port explicitly whatever its value is?

**Comment by tima** [comment]
@dvarrazzo are you using the `dest_port` param here or are you expecting the module to have a feature that recognizes the a src/dest is a remote and has a port in the URI?

**Comment by dvzubarev** [comment]
I'm using the following simple code

```
- synchronize:
    dest_port: 22
    src: "{{build_dir}}"
    dest: /tmp
```

I'm expecting that this `dest_port` will be used when connecting to a remote.
But instead of it, the port from /etc/ssh/ssh_config is in use (I have specified there Port 2222).
So I get `Connection refused` error.

**Comment by tima** [comment]
Hum. OK. I think I understand what's going on. Either the module has to parse the ssh_config to get the Port in the event its not 22 like in your case or the logic needs to be simplified to use 22 even if its already the default and is unnecessary. I'm leading towards the latter here.

**Comment by tima** [comment]
@dvzubarev would like to have you check out #4302 and tell us how @abadger's fix works out for you.

needs_info

**Comment by dvzubarev** [comment]
thank you, works for me!

**Comment by abadger** [comment]
Cool.  Fix has been merged to devel and the stable-2.1 branch.
